packages/ef-runtime-client/src/ComponentRegistry/ComponentRegistry.ts

```diff
@@ -0,0 +1,41 @@
+export interface IComponentRegistry {
+  fetch(systemCode: string): Promise<void>;
+  getURL(component: string): string | undefined;
+  getComponentKeys(): string[];
+  applyOverrides(overrides: { [propName: string]: string }): void;
+}
+
+export interface IComponentRegistryDependencies {
+  registryURL: string;
+}
+
+export class ComponentRegistry implements IComponentRegistry {
+  private registry: { [propName: string]: string } = {};
+  private registryURL: string;
+
+  constructor({ registryURL }: IComponentRegistryDependencies) {
+    this.registryURL = registryURL;
+  }
+
+  async fetch(systemCode: string): Promise<void> {
+    try {
+      const res = await fetch(`${this.registryURL}/?app=${systemCode}`);
+      const data = await res.json();
+      Object.assign(this.registry, data.imports);
+    } catch (err) {
+      console.error("Unable to fetch Component Registry", err);
+    }
+  }
+
+  getURL(component: string): string | undefined {
+    return this.registry[component];
+  }
+
+  getComponentKeys(): string[] {
```

Comment:
Instead of this can we just add `getRegistry` to get the registry object itself and the use of that can choose to get the keys or values of it.

Reply:
Hmmm yes I agree providing a getRegistry method could offer more flexibility, it may expose us to risks of unintended modifications to the registry object? Currently, methods like getURL, getComponentKeys, and applyOverrides allow us to encapsulate the logic and maintain more control over how the registry is accessed or modified. What are your thoughts on this trade-off?

Reply:
We can make the registry itself private and add a getter that only returns a copy of the registry so that we make sure no one can modify the registry object itself except through the `applyOverrides` method.